frontend/src/App.svelte

```diff
@@ -28,16 +28,37 @@
   $: if (query) landing = false;
 
   let ay2023 = false;
+  let genEdChecks: boolean[] = new Array(4).fill(false);
+  let genEdAreas: string[] = ["HSI", "STS", "EC", "AC"];
 
   const { data, error, search } = createSearcher();
-  $: finalQuery =
-    (ay2023 ? "@academicYear:[2023 2023] " : "") + normalizeText(query);
-  $: search(finalQuery);
+  let finalQuery = "";
+
+  // Search for GENED areas, if boxes are checked and "gened" remains in the search box. 
+  $: {
+    // Get | separated string of checked GENED areas. 
+    let stringTags: string[] = genEdChecks.map((checked, i) => checked ? genEdAreas[i] : "");
+    let genEdTagArr: string[] = []; 
+    stringTags.forEach(element => {
+      if (element) { genEdTagArr.push(element)}
+    });
+    let genEdSearchQuery = genEdTagArr.length ? " @genEdArea:{" + genEdTagArr.join("|") + "} " : ""
+
+    // Add tags to query, if boxes checked. 
+    finalQuery = genEdQuery ? 
+      (ay2023 ? "@academicYear:[2023 2023] " : "") + normalizeText(query) + genEdSearchQuery : 
+      (ay2023 ? "@academicYear:[2023 2023] " : "") + normalizeText(query); 
+
+    // If user wants only GENEDs. 
+    finalQuery = (query.indexOf("GENED") !== -1) ? finalQuery + " @subject:GENED " : finalQuery; 
```

Comment:
What's this line for?

Reply:
On my local setup, which has only a few years of Curricle data, searching `gened` will return 259 classes whereas the user likely only wants the 192 true GENED classes (there are about 60 other classes in e.g. genetics that get hit). I made searching "GENED" a way to only get gen-ed classes, but that behavior might not be clear.

(I do think having a gen-ed only search is useful though.)

Reply:
okay sounds good just use `genEdQuery` instead so that it works case-insensitively